## LAB 1 - Determining three Concrete Problems

### Step 1: Import and Set-up

In [5]:
#!pip install kagglehub -q

import os
import kagglehub
import pandas as pd

# Download latest version of the dataset
path = kagglehub.dataset_download("prasad22/healthcare-dataset")
print("Path to dataset files:", path)

# List files in the downloaded dataset folder
print(os.listdir(path))

Using Colab cache for faster access to the 'healthcare-dataset' dataset.
Path to dataset files: /kaggle/input/healthcare-dataset
['healthcare_dataset.csv']


### Step 1a: Loading the dataset into a DataFrame to inspect shape/columns

In [4]:
df = pd.read_csv(os.path.join(path, "healthcare_dataset.csv"))

print("Shape:", df.shape)
print("\nColumns:", df.columns.tolist())
print("\nFirst 5 rows:")
df.head()

Shape: (55500, 15)

Columns: ['Name', 'Age', 'Gender', 'Blood Type', 'Medical Condition', 'Date of Admission', 'Doctor', 'Hospital', 'Insurance Provider', 'Billing Amount', 'Room Number', 'Admission Type', 'Discharge Date', 'Medication', 'Test Results']

First 5 rows:


,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal


### Step 2: Basic data quality scan

### Step 2a: Checking Missing values per column

In [7]:
print("=== Missing values per column ===")
print(df.isnull().sum())

=== Missing values per column ===
Name                  0
Age                   0
Gender                0
Blood Type            0
Medical Condition     0
Date of Admission     0
Doctor                0
Hospital              0
Insurance Provider    0
Billing Amount        0
Room Number           0
Admission Type        0
Discharge Date        0
Medication            0
Test Results          0
dtype: int64


### Step 2b: Checking Dublicate rows

In [8]:
print("\n=== Duplicate rows ===")
print("Full duplicate rows:", df.duplicated().sum())


=== Duplicate rows ===
Full duplicate rows: 534


### Step 2c: Checking Quick stats on numeric columns

In [9]:
print("\n=== Quick stats on numeric columns ===")
df.describe()


=== Quick stats on numeric columns ===


,Age,Billing Amount,Room Number
count,55500.000000,55500.000000,55500.000000
mean,51.539459,25539.316097,301.134829
std,19.602454,14211.454431,115.243069
min,13.000000,-2008.492140,101.000000
25%,35.000000,13241.224652,202.000000
50%,52.000000,25538.069376,302.000000
75%,68.000000,37820.508436,401.000000
max,89.000000,52764.276736,500.000000


### Step 2d: Checking Data types

In [10]:
print("\n=== Data types ===")
print(df.dtypes)


=== Data types ===
Name                   object
Age                     int64
Gender                 object
Blood Type             object
Medical Condition      object
Date of Admission      object
Doctor                 object
Hospital               object
Insurance Provider     object
Billing Amount        float64
Room Number             int64
Admission Type         object
Discharge Date         object
Medication             object
Test Results           object
dtype: object


### Step 2e: Checking Text Consistency

In [12]:
print("=== Sample of Name, Doctor, Hospital columns (check capitalization) ===")
print(df[['Name', 'Doctor', 'Hospital']].head(10))

print("\n=== Unique values in categorical columns ===")
for col in ['Gender', 'Blood Type', 'Medical Condition', 'Insurance Provider', 'Admission Type', 'Test Results']:
    print(f"\n{col}: {df[col].unique()}")

=== Sample of Name, Doctor, Hospital columns (check capitalization) ===
                 Name            Doctor                     Hospital
0       Bobby JacksOn     Matthew Smith              Sons and Miller
1        LesLie TErRy   Samantha Davies                      Kim Inc
2         DaNnY sMitH  Tiffany Mitchell                     Cook PLC
3        andrEw waTtS       Kevin Wells   Hernandez Rogers and Vang,
4       adrIENNE bEll    Kathleen Hanna                  White-White
5       EMILY JOHNSOn     Taylor Newton               Nunez-Humphrey
6      edwArD EDWaRDs       Kelly Olson              Group Middleton
7  CHrisTInA MARtinez    Suzanne Thomas  Powell Robinson and Valdez,
8     JASmINe aGuIlaR   Daniel Ferguson                Sons Rich and
9    ChRISTopher BerG       Heather Day               Padilla-Walker

=== Unique values in categorical columns ===

Gender: ['Male' 'Female']

Blood Type: ['B-' 'A+' 'A-' 'O+' 'AB+' 'AB-' 'B+' 'O-']

Medical Condition: ['Cancer' 'Obesity'

### Step 2f: Checking Date Logic

In [13]:
print("\n=== Convert dates and check for discharge before admission ===")
df['Date of Admission'] = pd.to_datetime(df['Date of Admission'])
df['Discharge Date'] = pd.to_datetime(df['Discharge Date'])

bad_dates = df[df['Discharge Date'] < df['Date of Admission']]
print("Rows where discharge is before admission:", len(bad_dates))


=== Convert dates and check for discharge before admission ===
Rows where discharge is before admission: 0




---



### Three Concrete Problems identified are:
- 534 duplicate admission records
- Negative billing amounts
- Inconsistent naming